examples/server/server.cpp

```diff
@@ -71,7 +71,7 @@ struct llama_server_context
       {
         embd_inp.push_back(prompt_tokens[i]);
         if(new_prompt_len == 0) {
-          if(i - 1 < n_past) {
+          if(((int32_t)i) - 1 < n_past) {
```

Comment:
this almost looks like lisp
try `int32_t(i) - 1` instead

Reply:
Are the parenthesis actually even needed? I just did that to be safe because I wasn't sure if it meant `((type)i - 1)` or `((type)i) - 1`

Reply:
hm good point. took a look and `type(value)` syntax is c++, but pre c++11, so should be fine.

> Maybe the function style cast is just preferred in any case?

in my opinion yes. makes it more readable. but i also dont write lisp, so i might be biased :laughing:

Reply:
I'll just do it the way you suggested to make life easier for everyone.